src/threads.cc

```diff
@@ -0,0 +1,153 @@
+#include <dlfcn.h>
+#include <cstddef>
+#include <cstdlib>
+
+#include <iostream>
+
+#include <boost/algorithm/string/join.hpp> // for boost::algorithm::join
```

Comment:
boost::algorithm isn't mentioned on that page, but I think that's OK since it says "Most other Boost libraries may be used after appropriate design review."  It isn't an integral part of the implementation, so I _think_ it's OK if you say it's OK.  Checking with the Bosses...
I don't think the page needs to be updated, since it doesn't contain a list of all boost libraries we use.